docs/contributing/code-documentation.txt

```diff
@@ -286,3 +295,58 @@ Type Declaration
     # @option **kwargs [ String | Array<String> ] :items The items(s) as Strings to include.
     # @option **kwargs [ Integer ] :limit An Integer denoting the limit.
     def buy_groceries(**kwargs)
+
+- **Blocks:** Use ``@yield`` to specify when the method yields to a block.
+
+  .. code-block:: ruby
+
+    # @yield [ Symbol, Symbol, Symbol ] Evaluates the guess of who did the crime.
+    #   Must take the person, location, and weapon used. Must return true or false.
+    def whodunit
+      yield(:mustard, :ballroom, :candlestick)
+    end
+
+- **Blocks:** If the method explicitly specifies a block argument, include the
+  argument name in ``@yield`` preceded by an ampersand ``&``. Also use the ``@yield``
+  macro when the method uses ``block.call`` rather than ``yield`` internally.
```

Comment:
I don't see the purpose of this paragraph. `@yield` documents that the method yields, it doesn't matter that the block is taken as an explicit argument.

Reply:
@p-mongo If the discussion here is meant to be whether the behavior of a `&block` arg should be specified by `@param` or `@yield`, in [YARD itself](https://github.com/lsegal/yard/commit/50384a2bdcad6ffc90c9ecfd5877ab02bb9ab9b2) there is ambiguity and it seems that both ways are supported.

For my tastes, it seems more useful to always use `@yield`, `@yieldparam`, `@yieldreturn` to describe the shape of *how a passed-in block behaves*, and not get hung up over whether (a) the block is explicitly declares as a param `&block` and (b) the method is uses `yield` and `block.call` internally because in both (a) and (b) the end result is functionally equivalent.

Again: **this is a matter of preference.** My opinion here is that the code docs should focus on *semantic behavior*, and not a literal statement of `yield` vs. `block.call` appears in the code. In my view, if you want to know that, you can just read the code 😄 

If you'd like to do things differently, its your call, so just let me know what to do.